gateway/ln-gateway/src/bin/cln_extension.rs

```diff
@@ -109,7 +109,7 @@ pub struct ClnRpcService {
     client: Arc<Mutex<ClnRpc>>,
     interceptor: Arc<ClnHtlcInterceptor>,
     task_group: TaskGroup,
-    pub addr: SocketAddr,
+    listen: Option<options::Value>,
```

Comment:
We could probably avoid changing that type by just moving `ClnExtensionOpts::try_parse()` into `ClnRpcService::new`. I think that would make the code more elegant in general.

Reply:
oh nice